Common/Data/Custom/SmartInsider/SmartInsiderTransaction.cs

```diff
@@ -0,0 +1,374 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Globalization;
+using System.IO;
+
+namespace QuantConnect.Data.Custom.SmartInsider
+{
+    /// <summary>
+    /// Smart Insider Transaction - Contains information
+    /// about insider trading transactions
+    /// </summary>
+    public class SmartInsiderTransaction : SmartInsiderEvent
+    {
+        /// <summary>
+        /// Date traded through the market
+        /// </summary>
+        public DateTime? BuybackDate { get; private set; }
+
+        /// <summary>
+        /// Describes how transaction was executed
+        /// </summary>
+        public string BuybackVia { get; private set; }
+
+        /// <summary>
+        /// Describes which entity carried out the transaction
+        /// </summary>
+        public string BuybackBy { get; private set; }
```

Comment:
I'm curious if the usage of `private set` was decided upon or 'just happened' -- reviewing other data types in LEAN and it looks like most of them are all public get/set

Reply:
I'm a big fan of immutability, and in general wanted to avoid corruption of data in any way possible by avoiding external setters. Wasn't aware that public get/set was the convention, so I'll update it in my next commit 👍